benchopt/utils/conda_env_cmd.py

```diff
@@ -172,30 +172,44 @@ def delete_conda_env(env_name):
                capture_stdout=True)
 
 
-def install_in_conda_env(*packages, env_name=None, force=False, quiet=False):
-    """Install the packages with conda in the given environment"""
-    if len(packages) == 0:
-        return
+def get_cmd_from_requirements(packages):
+    """Process the packages from requirements and create the install cmd.
 
+    This detect the packages that needs to be installed with pip and also
+    the additional channels for conda packages.
+    """
     pip_packages = [pkg[4:] for pkg in packages if pkg.startswith('pip:')]
     conda_packages = [pkg for pkg in packages if not pkg.startswith('pip:')]
 
-    error_msg = ("Failed to conda install packages "
-                 f"{packages if len(packages) > 1 else packages[0]}\n"
-                 "Error:{output}")
     cmd = []
     if conda_packages:
-        packages = ' '.join(conda_packages)
-        cmd.append(f"{CONDA_CMD} install --update-all -y {packages}")
+        channels = ' '.join(set(
```

Comment:
Does this use common channels for all packages? And do we want to use some channels only for some packages (I'd say yes) ? So we'd need several "conda install" lines, each with its channel

Reply:
We want to have all installed packages in a same env, so usually what is done is to let the deps solver figure out which are compatible together. so I would not mess with multiple install as this tends to make the install unstable.
Moreover, this would make as many dependency resolution as you have lines so it would be slow.

Reply:
This is quite rare but it could happen. In this case, the most recent one will be installed.
But I think this is very marginal, we should not over engineer this. Most packages are in one channel (hopefully `conda-forge`) and we should only add a channel if the package is really not avail in the other channel IMO.

Reply:
then fine by me!